pubpub/pubpub
# feature: add content version field and preprint (posted_content) deposits

Closes #912

This PR adds a new field to the deposit form that allows a Pub admin to update the resource's `content_version`.

![Screen Shot 2020-07-31 at 4 32 19 PM](https://user-images.githubusercontent.com/6402908/89075092-6c3a7600-d34b-11ea-8647-05ebbf23f2dc.png)

The field's options are detailed below:

* **(None)** un-sets the `<resource>` element's `content_version` attribute of the work
* **Advance Manuscript** sets the `<resource>` element's `content_version` attribute to `"am"`(advance manuscript)
* **Version of Record** sets the `<resource>` element's `content_version` attribute to `"vor"`
* **Preprint** converts the record to a `posted_content` item

Since changing the content version of the work results in changes to the deposit, the entire deposit preview is re-fetched each time the dropdown's selected item changes, as demonstrated below:

<img src="https://user-images.githubusercontent.com/6402908/89075212-ad328a80-d34b-11ea-8a7f-e1c2554b1162.gif" width="550">

(sorry for the blurry gif)

**This PR does not include:**

- Automatically generating more specific deposits for peer reviews and supplementary material.
- Defaulting to preprint content version for Pubs with an `isPreprintOf` edge.

**Test Plan:**

1. Create a new Pub and publish it.
2. Click the "Preview Deposit" button in the Pub settings.
3. Toggle between the various options in the content version dropdown and verify they work as described above.
